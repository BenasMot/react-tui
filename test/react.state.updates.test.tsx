import React from "react";
import { getScreen } from "./fixtures/screen";
import { screenToString } from "./util/screen";
import { createTestRenderer, settle } from "./fixtures/render";
import { RootBox } from "./fixtures/components/RootBox";

it("<box /> updates", async () => {
  const screen = getScreen();
  const renderer = createTestRenderer(screen);
  const firstProps = {
    label: "first-label",
    border: { type: "line" },
  };
  const secondProps = {
    label: "second-label",
    border: { type: "bg", ch: "@" },
  };
  let setIsBoxVisible: any = () => null;
  let setProps: any = () => null;
  const UpdatingStyleBox = () => {
    const [props, _setProps] = React.useState(firstProps);
    const [isBoxVisible, _setIsBoxVisible] = React.useState(true);
    setIsBoxVisible = _setIsBoxVisible;
    setProps = _setProps; // gross.
    return <RootBox>{isBoxVisible && <box {...props} />}</RootBox>;
  };
  renderer.render(<UpdatingStyleBox />);
  // assert first paint OK
  await settle();
  expect(screenToString(screen)).toMatchSnapshot();
  // update props, assert second paint OK
  setProps(secondProps);
  await settle();
  expect(screenToString(screen)).toMatchSnapshot();
  // hide content, assert third paint OK
  setIsBoxVisible(false);
  await settle();
  expect(screenToString(screen)).toMatchSnapshot();
});
