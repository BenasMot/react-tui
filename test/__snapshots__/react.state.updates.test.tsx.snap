// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<box /> updates 1`] = `
"┌─root-box─────────────────────────────────────────────────┐
│┌─first-label────────────────────────────────────────────┐│
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
│└────────────────────────────────────────────────────────┘│
└──────────────────────────────────────────────────────────┘"
`;

exports[`<box /> updates 2`] = `
"┌─root-box─────────────────────────────────────────────────┐
│@@second-label@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@│
│@                                                        @│
│@                                                        @│
│@                                                        @│
│@                                                        @│
│@                                                        @│
│@                                                        @│
│@                                                        @│
│@                                                        @│
│@                                                        @│
│@                                                        @│
│@                                                        @│
│@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@│
└──────────────────────────────────────────────────────────┘"
`;

exports[`<box /> updates 3`] = `
"┌─root-box─────────────────────────────────────────────────┐
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
└──────────────────────────────────────────────────────────┘"
`;
